File: server/schemas/recipe.py
"""Pydantic schemas for recipe API responses."""

from pydantic import BaseModel


class IngredientResponse(BaseModel):
    """Ingredient within a recipe."""

    name: str
    amount: str


class InstructionResponse(BaseModel):
    """Single cooking instruction."""

    step: int
    description: str


class RecipeResponse(BaseModel):
    """Complete recipe returned to the frontend."""

    id: int
    title: str
    description: str
    image: str
    categories: list[str]
    cookTimeMinutes: int
    servings: int
    ingredients: list[IngredientResponse]
    instructions: list[InstructionResponse]

File: frontend/src/types/recipe.ts
export interface Ingredient {
  name: string
  amount: string
}

export interface Instruction {
  step: number
  description: string
}

export interface Recipe {
  id: number
  title: string
  description: string
  image: string
  categories: string[]
  cookTimeMinutes: number
  servings: number
  ingredients: Ingredient[]
  instructions: Instruction[]
}

export interface CategoryInfo {
  name: string
  count: number
}

export interface SearchRequest {
  query: string
  filters?: string[]
  limit?: number
}

export interface SearchResponse {
  query: string
  total: number
  recipes: Recipe[]
}

export interface SearchSuggestion {
  text: string
  source: string
}
